### ISSUE TITLE
Use phi to resolve caches after an if/else block

### ISSUE THREAD

**Issue by AjaniBilby** [qupa-lang/Qupa#37]
```qupa
int a = 4;
if (rand() % 2 == 0) {
	a = 3;
} else {
	a = 6;
}
```
After an if/else statement variable caches could be in two possible states (i.e. above), hence a ``phi`` should be used to resolve the possible states rather than having to reload the value.

**Comment by AjaniBilby** [comment]
The produced LLVM would be something along the lines of:
```llvm
0:
	// presume %1 hold the result of rand() % 2 == 0
	br i1 %1, label %2, label %X
2:
	br label %X
3:
	br label %X
4:
	%5 = phi i32 [ 3, %2 ], [ 6, %3 ]
```
Note the final line is simply to show the value resolving, even though the value is not later used in the code example.

**Comment by AjaniBilby** [comment]
Implemented in #47
